AppWidget/app/src/main/java/com/example/android/appwidget/ItemsCollectionAppWidget.kt

```diff
@@ -0,0 +1,185 @@
+        ): RemoteViews {
+            val remoteViews = RemoteViews(context.packageName, layoutId)
+            if (BuildCompat.isAtLeastS()) {
+                // Compound buttons in a widget are stateless. You need to change the state and register for
+                // the state change events.
+                when (layoutId) {
+                    R.layout.item_checkboxes -> {
+                        // This code will check the Checkbox
+                        remoteViews.setCompoundButtonChecked(R.id.item_checkbox, true)
+                    }
+                    R.layout.item_radio_buttons -> {
+                        // This code will check the item_radio_button2 in the item_radio_group RadioGroup
+                        remoteViews.setRadioGroupChecked(
+                            R.id.item_radio_group,
+                            R.id.item_radio_button2
+                        )
+                    }
+                    R.layout.item_switches -> {
+                        val viewId = R.id.item_switch
+                        val onCheckedChangePendingIntent = PendingIntent.getBroadcast(
+                            context,
+                            REQUEST_CODE_FROM_COLLECTION_WIDGET,
+                            Intent(context, ItemsCollectionAppWidget::class.java).apply {
+                                putExtra(EXTRA_VIEW_ID, viewId)
+                                putExtra(REQUEST_CODE, REQUEST_CODE_FROM_COLLECTION_WIDGET)
+                            },
+                            // API level 31 requires specifying either of
+                            // PendingIntent.FLAG_IMMUTABLE or PendingIntent.FLAG_MUTABLE
+                            PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_MUTABLE
+                        )
+                        // Listen for change events.
+                        // RemoteResponse.fromPendingIntent works on an individual item whereas you can set
+                        // a PendingIntent template using RemoteViews.setPendingIntentTemplate and
+                        // distinguish individual on-click by calling RemoteResponse.fromFillInIntent.
+                        // See
+                        // https://developer.android.com/reference/android/widget/RemoteViews.RemoteResponse#fromPendingIntent(android.app.PendingIntent)
+                        // https://developer.android.com/reference/android/widget/RemoteViews.RemoteResponse#fromFillInIntent(android.content.Intent)
+                        // for more details.
+                        remoteViews.setOnCheckedChangeResponse(
+                            viewId,
+                            RemoteViews.RemoteResponse.fromPendingIntent(
+                                onCheckedChangePendingIntent
+                            )
+                        )
+                    }
+                }
+            }
+            return remoteViews
+        }
+    }
+
+    override fun onCreate() {}
```

Comment:
Potentially silly question: Do these need to be overridden? We're not actually implementing any functionality in these method overrides.

Reply:
They are functions from RemoteViewsFactory. So overriding these are needed.